python/gui/MaaS/static/common/js/domain.js

```diff
@@ -0,0 +1,206 @@
+
```

Comment:
Something I'm seeing that you should try is to use `let` instead of `var`, declare one variable per statement, avoid acronyms (such as `i` and `l`), use `for (let x of y)` statements instead of classical for loops, and maybe take advantage of jQuery more when it comes to iterating over collections of stuff like `option`s.

Don't sleep on `.forEach` on lists either. Super useful.

Reply:
I've gone through and changed the multi-variable `var`s to single variable `let`s, along with a few small tweaks to variable names.

Reply:
Feel comfortable adding some jsdoc? I understand if you're running to get all this in there so you can move on, but if you feel it's reasonable, jsdoc entries help with stuff like type hinting and will make it easier for others to work on this in the future.